cloud/client.go

```diff
@@ -339,17 +128,19 @@ func NewClient(host, agentSockPath, authCredsOverride string, warnFunc func(stri
 			return nil, err
 		}
 	}
-	return c, nil
-}
-
-func getMessageFromJSON(r io.Reader) (string, error) {
-	decoder := json.NewDecoder(r)
-	msg := struct {
-		Message string `json:"message"`
-	}{}
-	err := decoder.Decode(&msg)
+	tlsConfig := credentials.NewTLS(&tls.Config{})
+	ctx := context.Background()
+	retryOpts := []grpc_retry.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(100 * time.Millisecond)),
+		grpc_retry.WithCodes(codes.Internal, codes.Unavailable),
+	}
```

Comment:
I'm a little worried about these retry options being built into every call.
Unless we can ensure that every call will always be idempotent.
So I think we should either:

 - Leave this off the default dial options and expose these as defaults we purposefully use per RPC. I like this one because it makes it very obvious how the call will behave.
 - Do something struct embedding and implement specific calls on each struct depending on idempotency.

Reply:
To be honest, I was just trying to roughly replicate the retry logic that was already here when these were HTTP calls. 

Ref: https://github.com/earthly/earthly/blob/3111d90507ab09de8aded484300ea3d7e207739f/cloud/client.go#L162

I assumed the reason these retries were coded in the first place was just in case there was some transient error, and we didn't want to bother the user with it if a simple retry would solve it.

As for idempotency - I don't think we need to worry too much about it. I don't think it's a big deal if these operations are re-run on a failure, and in fact the user would probably just re-run it themselves anyway. If we have an operation that is risky if it's not idempotent (e.g. charging a credit card) we should make sure its safe on the backend (we have the tools necessary with DB rollbacks, request-IDs, etc)

In general, I'd sooner just get rid of this retry logic than let it get too complicated. think it might be nice to have, in the odd time that it actually kicks in and works, but the old custom retry code (at least in my experience) was a bit difficult to maintain and a good source for coding mistakes.

Reply:
Mike, Corey, Brandon, and I had a discussion around this in a Zoom meeting and we came to the  following conclusion:
 - The old retry logic was complex and we're glad to replace it.
 - Risk vs Reward: we believe it's more important to have retries and for the client to always handle that by default rather than adding call options as the third argument to every gRPC call.  i.e.,
 ```
 resp, err := c.pipelines.ListSatellites(c.withAuth(ctx), &pb.ListSatellitesRequest{
		OrgId: orgID,
	}, ...defaultRetryOptions())
```
 - If we have an operation we are worried about in the future we will make an exception then.
 - We will aim to try our best to handle these scenarios on the server side through rate limiting and idempotency.